### ISSUE TITLE
[a11y]: Composed Modal, Dialog heading is provided in h3 instead of h2

### ISSUE THREAD

**Issue by anamikaanu96** [carbon-design-system/carbon#17397]
### Package

@carbon/react

### Browser

Firefox

### Operating System

Windows

### Package version

1.61.0

### Automated testing tool and ruleset

manual

### Assistive technology

JAWS

### Description

Maintain hierarchical order for the Heading levels :
Recommended to present the Modal Main heading in H2.

### WCAG 2.1 Violation

1.3.1 Info & Relationships

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-composedmodal--default

### Steps to reproduce

storybook : https://react.carbondesignsystem.com/?path=/story/components-composedmodal--default

### Suggested Severity

None

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
